nimare/workflows/cbma.py

```diff
@@ -0,0 +1,177 @@
+        If estimator, corrector, or diagnostics are passed as initialized objects, this parameter
+        will be ignored.
+        Default is 1.
+
+    Returns
+    -------
+    :obj:`~nimare.results.MetaResult`
+        Results of Estimator and Corrector fitting with cluster and diagnostic tables.
+    """
+    n_cores = _check_ncores(n_cores)
+
+    if not isinstance(diagnostics, list) and diagnostics is not None:
+        diagnostics = [diagnostics]
+
+    # Check dataset type
+    dataset = _check_type(dataset, Dataset)
+
+    # Options allows for string input
+    estm_options = ("ale", "scale", "mkdadensity", "kda")
+    corr_options = ("montecarlo", "fdr", "bonferroni")
+    diag_options = ("jackknife", "focuscounter")
+
+    # Check inputs and set defaults if input is None
+    estimator = (
+        ALE(n_cores=n_cores)
+        if estimator is None
+        else _check_input(estimator, CBMAEstimator, estm_options, n_cores=n_cores)
+    )
+    corrector = (
+        FWECorrector(method="montecarlo", n_cores=n_cores)
+        if corrector is None
+        else _check_input(corrector, Corrector, corr_options, n_cores=n_cores)
+    )
+
+    pval = 0.05 if corrector.method == "montecarlo" else 0.001
+    vthr = p_to_z(pval, tail="one")  # Set voxel_thresh for diagnostics
+    if diagnostics is None:
+        diagnostics = [Jackknife(voxel_thresh=vthr, n_cores=n_cores)]
+    else:
+        diagnostics = [
+            _check_input(diagnostic, Diagnostics, diag_options, voxel_thresh=vthr, n_cores=n_cores)
+            for diagnostic in diagnostics
+        ]
+
+    if isinstance(estimator, PairwiseCBMAEstimator):
+        raise AttributeError(
+            'The "cbma_workflow" function does not currently work with pairwise Estimators.'
+        )
+
+    LGR.info("Performing meta-analysis...")
+    results = estimator.fit(dataset)
+
+    LGR.info("Performing correction on meta-analysis...")
+    corr_results = corrector.transform(results)
+
+    LGR.info("Generating clusters tables and performing diagnostics on corrected meta-analyses...")
+    img_keys = [
+        img_key
+        for img_key in corr_results.maps.keys()
+        if img_key.startswith("z_") and ("_corr-" in img_key)
+    ]
+    for img_key, diagnostic in itertools.product(img_keys, diagnostics):
+        diagnostic.target_image = img_key
+        contribution_table, clusters_table, _ = diagnostic.transform(corr_results)
+
+        diag_name = diagnostic.__class__.__name__
+        clust_tbl_name = f"{img_key}_clust"
+        count_tbl_name = f"{img_key}_{diag_name}"
+        if not clusters_table.empty:
+            corr_results.tables[clust_tbl_name] = clusters_table
+            corr_results.tables[count_tbl_name] = contribution_table
+        else:
+            LGR.warning(
+                f"Key {count_tbl_name} and {clust_tbl_name} will not be stored in "
+                "MetaResult.tables dictionary."
+            )
```

Comment:
I feel like we had a discussion about this, where I agreed that not saving empty tables was good behavior.

But now I feel like I'm flip flopping on that decision. If the output is valid (i.e., the data and code are correct, it's just the data wasn't signifiicant) then I think I should have the cluster table and contribution table saved, even if they are empty.

Reply:
The `None` for the contribution table may rise an error when saving the `.tsv`. What do you think we should do in that case?

Reply:
Raise a warning when the table is None.